contracts/cw4973/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+pub use crate::error::ContractError;
+pub use crate::msg::{ExecuteMsg, InstantiateMsg};
+pub use crate::state::{ADR36SignDoc, Fee, MsgSignData, MsgSignDataValue, PermitSignature};
+use cosmwasm_std::{
+    entry_point, Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response, StdResult,
+};
+pub use cw721_base::{
+    entry::{execute as _execute, query as _query},
+    state::TokenInfo,
+    ContractError as Cw721ContractError, Cw721Contract, ExecuteMsg as Cw721BaseExecuteMsg,
+    Extension, InstantiateMsg as Cw721BaseInstantiateMsg, MintMsg, MinterResponse,
+    QueryMsg as Cw721QueryMsg,
+};
+
+use bech32::{ToBase32, Variant::Bech32};
+use ripemd::Ripemd160;
+use sha2::{Digest, Sha256};
+use std::{env, str};
+
+pub mod error;
+pub mod msg;
+pub mod state;
+pub mod test;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:cw4973";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+const AGREEMENT_STRING: &str =
+    "Agreement(string chain_id,address active,address passive,string tokenURI)";
+
+pub type Cw4973Contract<'a> = Cw721Contract<'a, Extension, Empty, Empty, Empty>;
+pub type QueryMsg = cw721_base::QueryMsg<Empty>;
+#[cfg(not(feature = "library"))]
+pub mod entry {
+    use super::*;
+
+    #[entry_point]
+    pub fn instantiate(
+        mut deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: InstantiateMsg,
```

Comment:
what do you think about making the `InstantiateMsg` an alias for `Cw721BaseInstantiateMsg`([ex](https://github.com/DA0-DA0/dao-contracts/blob/8d7ad103c9c4bc3a2477a4d4f50532ef573f4d83/contracts/external/dao-migrator/src/msg.rs#L14-L16))?

```
type InstantiateMsg = Cw721BaseInstantiateMsg;
```

that would remove the need for the first line of the function and imo helps convey the intent a little better.